museek/plugin/rawdata_flagger_plugin.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+import numpy as np
+from matplotlib import pyplot as plt
+
+from ivory.plugin.abstract_plugin import AbstractPlugin
+from ivory.utils.requirement import Requirement
+from ivory.utils.result import Result
+from museek.enums.result_enum import ResultEnum
+from museek.factory.data_element_factory import FlagElementFactory
+from museek.flag_list import FlagList
+from museek.time_ordered_data import TimeOrderedData
+from museek.visualiser import waterfall
+
+
+class RawdataFlaggerPlugin(AbstractPlugin):
+    """ Plugin to flag raw data with values below a minimum """
+
+    def __init__(self,
+                 flag_minimum: float):
+        """
+        Initialise
+        :param flag_minimum: lower threshold to flag the data
+        """
+        super().__init__()
+        self.data_element_factory = FlagElementFactory()
+        self.flag_minimum = flag_minimum
+
+    def set_requirements(self):
+        """ Set the requirements. """
+        self.requirements = [Requirement(location=ResultEnum.DATA, variable='data'),
+                             Requirement(location=ResultEnum.OUTPUT_PATH, variable='output_path')]
+
+    def run(self, data: TimeOrderedData, output_path: str):
+        """
+        Flag raw data with values below a minimum
+        :param data: time ordered data of the entire block
+        :param output_path: path to store results
+        """
+        data.load_visibility_flags_weights()
+        new_flag = np.zeros(data.shape, dtype=bool)
```

Comment:
it's safer to do `data.visibility.shape`
In this case it's the same, but `data.shape` is always the complete data shape and `data.visibility.shape` is only the part that's loaded, e.g. only the scan part or only the calibrator observation.